src/MultiSigSigner.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {IDelegation} from "./interfaces/IDelegation.sol";
+import {ISigner} from "./interfaces/ISigner.sol";
+
+contract MultiSigSigner is ISigner {
+    ////////////////////////////////////////////////////////////////////////
+    // Constants
+    ////////////////////////////////////////////////////////////////////////
+
+    /// @dev The magic value returned by `isValidSignatureWithKeyHash` when the signature is valid.
+    /// - Calcualated as: bytes4(keccak256("isValidSignatureWithKeyHash(bytes32,bytes32,bytes)")
+    bytes4 internal constant MAGIC_VALUE = 0x8afc93b4;
+
+    /// @dev The magic value returned by `isValidSignatureWithKeyHash` when the signature is invalid.
+    bytes4 internal constant FAIL_VALUE = 0xffffffff;
+
+    ////////////////////////////////////////////////////////////////////////
+    // Errors
+    ////////////////////////////////////////////////////////////////////////
+
+    /// @dev The threshold can't be zero.
+    error InvalidThreshold();
+
+    /// @dev The owner key hash was not found in the config.
+    error OwnerNotFound();
+
+    /// @dev The key hash is invalid.
+    error InvalidKeyHash();
+
+    /// @dev Multisigs cannot be re-initialized.
+    error ConfigAlreadySet();
+
+    ////////////////////////////////////////////////////////////////////////
+    // Storage
+    ////////////////////////////////////////////////////////////////////////
+
+    struct Config {
+        uint256 threshold;
+        bytes32[] ownerKeyHashes;
+    }
+
+    /// @dev A config is mapped to a tuple of (address, keyhash)
+    /// This allows a single account, to register multiple multi-sig configs.
+    mapping(address => mapping(bytes32 => Config)) internal _configs;
+
+    function getConfig(address account, bytes32 keyHash)
+        public
+        view
+        returns (uint256 threshold, bytes32[] memory ownerKeyHashes)
+    {
+        Config memory config = _configs[account][keyHash];
+        return (config.threshold, config.ownerKeyHashes);
+    }
+
+    ////////////////////////////////////////////////////////////////////////
+    // Config Functions
+    ////////////////////////////////////////////////////////////////////////
+
+    function _checkKeyHash(bytes32 expectedKeyHash) internal view {
+        bytes32 keyHash = IDelegation(msg.sender).getContextKeyHash();
+        if (keyHash != expectedKeyHash) revert InvalidKeyHash();
+    }
+
+    function setConfig(bytes32 keyHash, uint256 threshold, bytes32[] memory ownerKeyHashes)
+        public
+    {
+        // Threshold can't be zero
+        if (threshold == 0) revert InvalidThreshold();
+
+        Config storage config = _configs[msg.sender][keyHash];
+
+        if (config.threshold > 0) {
+            revert ConfigAlreadySet();
+        }
+
+        _configs[msg.sender][keyHash] =
+            Config({threshold: threshold, ownerKeyHashes: ownerKeyHashes});
+    }
```

Comment:
Prefer renaming to `initConfig`, to highlight that this is not an update function.